### PR TITLE
Replace step comments with echo to avoid confusion

### LINKED ISSUE #7640: [BUG] classification recipe fails to run due to Makefile comments
### Issues Policy acknowledgement

- [X] I have read and agree to submit bug reports in accordance with the [issues policy](https://www.github.com/mlflow/mlflow/blob/master/ISSUE_POLICY.md)

### Willingness to contribute

Yes. I can contribute a fix for this bug independently.

### MLflow version

- Client: 2.1.1

### System information

- **Win10**:
- **3.10.8**:
- **GNU Make 3.81, installed via "Winget install GnuWin32.make"**:

### Describe the problem

I have tried the classification example from here, running with Win10 and the manually installed Make 3.81:
https://github.com/mlflow/recipes-examples

### Tracking information

_No response_

### Code to reproduce issue

- run the notebook recipes-expamples/classification/notebook/jupyter.ipynb 
- the split step fails due to this error:

```shell
process_begin: CreateProcess(NULL, # Run MLFlow Recipe step: split, ...) failed.
make (e=2): The system cannot find the file specified.

make: *** [steps/split/outputs/validation.parquet] Error 2
```

As it appears, the error is caused by the line comments starting in each of the blocks of the generated Makefile, here the step block:
# Run MLFlow Recipe step: split

as written in the file mlflow/recipes/utils/execution.py, line 527
(as well as in the other blocks).
In addition, some relative paths for the targets are filled with % in between, which causes issues.

These fixes seem to fix the issues and make the jupyter notebook to run:
https://github.com/mlflow/mlflow/commit/bdea43d9a657eeaca7b302aad57ce81010710278

It would be nice to fix this.

### Stack trace

```shell
process_begin: CreateProcess(NULL, # Run MLFlow Recipe step: split, ...) failed.
make (e=2): The system cannot find the file specified.

make: *** [steps/split/outputs/validation.parquet] Error 2
```

### Other info / logs

_No response_

### What component(s) does this bug affect?

- [ ] `area/artifacts`: Artifact stores and artifact logging
- [ ] `area/build`: Build and test infrastructure for MLflow
- [ ] `area/docs`: MLflow documentation pages
- [ ] `area/examples`: Example code
- [ ] `area/model-registry`: Model Registry service, APIs, and the fluent client calls for Model Registry
- [ ] `area/models`: MLmodel format, model serialization/deserialization, flavors
- [X] `area/recipes`: Recipes, Recipe APIs, Recipe configs, Recipe Templates
- [ ] `area/projects`: MLproject format, project running backends
- [ ] `area/scoring`: MLflow Model server, model deployment tools, Spark UDFs
- [ ] `area/server-infra`: MLflow Tracking server backend
- [ ] `area/tracking`: Tracking Service, tracking client APIs, autologging

### What interface(s) does this bug affect?

- [ ] `area/uiux`: Front-end, user experience, plotting, JavaScript, JavaScript dev server
- [ ] `area/docker`: Docker use across MLflow's components, such as MLflow Projects and MLflow Models
- [ ] `area/sqlalchemy`: Use of SQLAlchemy in the Tracking Service or Model Registry
- [X] `area/windows`: Windows support

### What language(s) does this bug affect?

- [ ] `language/r`: R APIs and clients
- [ ] `language/java`: Java APIs and clients
- [ ] `language/new`: Proposals for new client languages

### What integration(s) does this bug affect?

- [ ] `integrations/azure`: Azure and Azure ML integrations
- [ ] `integrations/sagemaker`: SageMaker integrations
- [ ] `integrations/databricks`: Databricks integrations

### PR DESCRIPTION
**Pull request by harupy**
Signed-off-by: harupy <hkawamura0130@gmail.com>

## Related Issues/PRs

 #7640

## What changes are proposed in this pull request?

This replace  step comments with echo to avoid confusion and fixes #7640.

## How is this patch tested?

- [x] Existing unit/integration tests
- [ ] New unit/integration tests
- [ ] Manual tests (describe details, including test results, below)

## Does this PR change the documentation?

- [ ] No. You can skip the rest of this section.
- [ ] Yes. Make sure the changed pages / sections render correctly in the documentation preview.

### Is this a user-facing change?

- [ ] No. You can skip the rest of this section.
- [ ] Yes. Give a description of this change to be included in the release notes for MLflow users.

(Details in 1-2 sentences. You can just refer to another PR with a description if this PR is part of a larger change.)

### What component(s), interfaces, languages, and integrations does this PR affect?

Components

- [ ] `area/artifacts`: Artifact stores and artifact logging
- [ ] `area/build`: Build and test infrastructure for MLflow
- [ ] `area/docs`: MLflow documentation pages
- [ ] `area/examples`: Example code
- [ ] `area/model-registry`: Model Registry service, APIs, and the fluent client calls for Model Registry
- [ ] `area/models`: MLmodel format, model serialization/deserialization, flavors
- [x] `area/recipes`: Recipes, Recipe APIs, Recipe configs, Recipe Templates
- [ ] `area/projects`: MLproject format, project running backends
- [ ] `area/scoring`: MLflow Model server, model deployment tools, Spark UDFs
- [ ] `area/server-infra`: MLflow Tracking server backend
- [ ] `area/tracking`: Tracking Service, tracking client APIs, autologging

Interface

- [ ] `area/uiux`: Front-end, user experience, plotting, JavaScript, JavaScript dev server
- [ ] `area/docker`: Docker use across MLflow's components, such as MLflow Projects and MLflow Models
- [ ] `area/sqlalchemy`: Use of SQLAlchemy in the Tracking Service or Model Registry
- [ ] `area/windows`: Windows support

Language

- [ ] `language/r`: R APIs and clients
- [ ] `language/java`: Java APIs and clients
- [ ] `language/new`: Proposals for new client languages

Integrations

- [ ] `integrations/azure`: Azure and Azure ML integrations
- [ ] `integrations/sagemaker`: SageMaker integrations
- [ ] `integrations/databricks`: Databricks integrations

<a name="release-note-category"></a>

### How should the PR be classified in the release notes? Choose one:

- [ ] `rn/breaking-change` - The PR will be mentioned in the "Breaking Changes" section
- [x] `rn/none` - No description will be included. The PR will be mentioned only by the PR number in the "Small Bugfixes and Documentation Updates" section
- [ ] `rn/feature` - A new user-facing feature worth mentioning in the release notes
- [ ] `rn/bug-fix` - A user-facing bug fix worth mentioning in the release notes
- [ ] `rn/documentation` - A user-facing documentation change worth mentioning in the release notes

### CODE REVIEW

**Review by dbczumar — APPROVED** [review]
LGTM! Thanks @harupy ! (Let's remove the comments).

cc @sunishsheth2009 can you file an internal ticket to track adding test coverage for Win32 GNU Make?

File: mlflow/recipes/utils/execution.py [hunk 1]
```diff
@@ -516,15 +516,15 @@ def __format__(self, path_spec: str) -> str:
 # ensuring that data is only ingested for downstream steps if it is not already present on the
 # local filesystem
 steps/ingest/outputs/dataset.parquet: steps/ingest/conf.yaml {path:prp/steps/ingest.py}
-	# Run MLFlow Recipe step: ingest
+    # Run MLFlow Recipe step: ingest
```
Comment (harupy) [inline mlflow/recipes/utils/execution.py:519]: Our step caching logic depends on those comments.

**Review by BenWilson2 — APPROVED** [review]
LGTM!

File: mlflow/recipes/utils/execution.py [hunk 2]
```diff
@@ -516,15 +516,15 @@ def __format__(self, path_spec: str) -> str:
 # ensuring that data is only ingested for downstream steps if it is not already present on the
 # local filesystem
 steps/ingest/outputs/dataset.parquet: steps/ingest/conf.yaml {path:prp/steps/ingest.py}
-	# Run MLFlow Recipe step: ingest
+	echo "Run MLFlow Recipe step: ingest"
```
Comment (harupy) [inline mlflow/recipes/utils/execution.py:519]: @floschl After discussion, we decided to use `echo`. Would mind testing this?

**Pull request merged**